### PR TITLE
BRITECHSRM-289 Import due dates as event dates

### PR DESCRIPTION
**Pull request by jebbinSCL**
Add fetch and conversion of Due dates to event dates on each session. 

Do not try and run import! ApiClient does not match up current api yet. 

Also Import will currently fail due to data issues in Trello. These will be dealt with soon.

### CODE REVIEW

File: TrelloSync/TrelloSync/App.config [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6" />
-    </startup>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6" />
+  </startup>
   <appSettings file="TrelloCreds.config">
-    <add key ="TrelloKey" value="UNSET" />
-    <add key ="TrelloToken" value="UNSET" />
-    <add key ="UseCache" value ="true"/>
+    <add key ="TrelloKey" value ="UNSET" />
```
Comment (pallenSL) [inline TrelloSync/TrelloSync/App.config]: Magic strings!

File: TrelloSync/TrelloSync/Config.fs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+open System.Configuration
+open Serilog
+
+let getBoolConfig (key : string) =
+    match bool.TryParse <| ConfigurationManager.AppSettings.Item(key) with
+    | true , value -> value
+    | _ -> 
+        let message = sprintf "Could not parse configuration value for key: %s as boolean" key
+        Log.Fatal(key)
```
Comment (pallenSL) [inline TrelloSync/TrelloSync/Config.fs]: Do you mean Log.Fatal(message) ?

File: TrelloSync/TrelloSync/Program.fs [hunk 3]
```diff
@@ -1,27 +1,17 @@
...
-open System.IO
-open System.Configuration
-open Logging
+﻿open Logging
 open Serilog
+open System.IO
 
 [<EntryPoint>]
 let main _ = 
+    JsonSettings.setDefaults()
```
Comment (pallenSL) [inline TrelloSync/TrelloSync/Program.fs]: We probably want all on main within try, with ex -> Log.Fatal(), exit 1

Comment (jebbinSCL) [inline TrelloSync/TrelloSync/Program.fs]: I really dislike that code, but I'll add it.

Comment (pallenSL) [inline TrelloSync/TrelloSync/Program.fs]: Where did you think that exception you throw in Config.fs was going to show up?

Comment (jebbinSCL) [inline TrelloSync/TrelloSync/Program.fs]: It was already logged there. This code means that any exception that is already logged will be logged twice. Also, a non zero exit code is returned automatically if an uncaught exception is thrown.

Comment (jebbinSCL) [inline TrelloSync/TrelloSync/Program.fs]: Added the change.

File: TrelloSync/TrelloSync/Program.fs [hunk 4]
```diff
@@ -4,24 +4,29 @@ open System.IO
...
-        else 0
-    
-    0
+        let result = 
+            if Config.performImport then Import.importAll trelloData
+            else 0
+        0
+    with
+        | ex -> 
+            Log.Fatal(ex.Message)
```
Comment (pallenSL) [inline TrelloSync/TrelloSync/Program.fs]: It is better to log "ex" itself, and try it to make sure you do get all the info - F# seems to like using InnerException.

**Comment by pallenSL** [comment]
:shipit:

**Pull request merged**